### ISSUE TITLE
More Advice and Bug Reporting for Chemical Werewolf

### ISSUE THREAD

**Issue by Sunev** [uhyo/jinrou#25]
Advices:
1. Always, FrankensteinsMonster absorbs someone's main job with a tip like “〇〇 absorbed the skill of TroubleMaker from ××”,while in Chemical Werewolf we found that tip becomes “〇〇 absorbed the skill of TroubleMaker×HolyMarked from ××”. Actually, the FrankensteinsMonster only absorbed the skill of TroubleMaker, the main job. Would you like to take this tip as kind of spoiler?

Bugs:
1.     Lonely Friend still exists. It seems that not only occurs when BadLady proposed to someone. In [this room](http://www.werewolf.online/room/26833)  (Chinese), 2 Cupids created 2 pairs of Friends separately, one pair acts normally and the other pair not.
2.  Devil×sth.
   Game never ends while Devil×sth achieved all his Victory Conditions. Even though x was bitten by werewolf and hanged at the next sunset. With `flag` not defined in `class Chemical`.

```
if @players.some((x)->x.isJobType "Devil" && x.flag=="winner" && x.team=="Devil")
    team="Devil"
```

```
x.isJobType "Devil" #   true
x.flag=="winner" #  false
```
1.  Fugitive×sth.
   Fugitive×sth won't die when run to some Target who was bitten by werewolf or Vampire. Because `x.target`  is undefined.

```
runners=game.players.filter (x)=>!x.dead && x.isJobType("Fugitive") && x.target==t.id
```

```
x.isJobType "Fugitive"       #  true
x.target==t.id #false
```
1.  Ushinotokimairi×sth.
   ushinotokimairi_curse acts randomly, as `midnightSort` didn't got redefined in `class Chemical`. `midnightSort` of Ushinotokimairi×sth is `100` as default instead of `90` for Ushinotokimairi. And we can imagine the mess of all other Jobs whose `midnightSort` is not 100.
2.  Sth×ToughGuy.
   At the next sunset after bitten by werewolf, ToughGuy, the subjob, `setDead` himself normally, while the mainjob Sth and the top Object class Chemical doesn't. Sth×ToughGuy becomes Walking Dead. The Heaven is visible to him. And what’s worse, ToughGuy `@setFlag null` for himself, Sth×ToughGuy becomes Invincible to Werewolf.
   Example:

```
{
    "type": "Complex",
    "dead": false,
    "flag": null
    "Complex_main": {
        "type": "Couple",
        "dead": false,
        "flag": null
    },
    "Complex_sub": {
        "type": "ToughGuy",
        "dead": true,
        "flag": null
    },
    "Complex_type": "Chemical",
    "Complex_flag": null
}
```

**Comment by Sunev** [comment]
Bug No.2 (the Devil things) also ooccurs in normal rooms, e.g. Devil with DeathNote, Devil was Counseled and so on.

**Comment by Sunev** [comment]
The midnightSort things mentioned in this issue has been fixed in #34 .

**Comment by Sunev** [comment]
About Fugitive×sth things, now Fugitive could die normally by using ```.accessByJobType("Fugitive").target```(#31 ) when he run to some Target who was bitten by Werewolf or Vampire, and we can protect Fugitive he himself from Werewolf and Vampire by making some change about how to deal with WalkingDead(#27 ).

But I cannot figure out wether this is an elegant solution, and it will conflict with your document "逃亡者などの襲撃耐性は吸収できません"(http://jinrou-doc.uhyohyo.net/jobs/FrankensteinsMonster.html). What is your view of point about it?
